# 10 Forecast

On holdout months, does the pre-2020 mean for that calendar month beat a forecast of a zero monthly log return?

Holdout starts 2020-01-01. Verdict: held.

This study reads stored series. It does not add a derived column.

In [1]:
import json
from pathlib import Path
studies = json.loads(Path('data/studies.json').read_text())
study = next(item for item in studies['studies'] if item['slug'] == 'forecast')
print(study['verdict'])
print(study['verdict_line'])
print(study['table']['caption'])
for row in study['table']['rows']:
    print(' | '.join(row))


held
Holdout months: 81. Mean absolute error of the train calendar-month mean is 10.162%. Mean absolute error of a flat zero forecast is 10.365%. The difference is 0.204%. The calendar mean is lower error, which was the bar.
Holdout monthly RWTC log return, mean absolute error
Train mean for that calendar month | 81 | 10.162%
Flat zero | 81 | 10.365%
TimesFM median, one month ahead | 81 | 10.535%


## Method

Mean absolute error on holdout months decides it. The seasonal forecast has to be strictly lower error than zero.

## Result

Holdout months: 81. Mean absolute error of the train calendar-month mean is 10.162%. Mean absolute error of a flat zero forecast is 10.365%. The difference is 0.204%. The calendar mean is lower error, which was the bar.